### DISCUSSION TITLE
Unable to resolve service for type 'YesSql.ISession' while attempting to activate 'OrchardCore.Workflows.Http.Services.HttpRequestRouteActivator'

### DISCUSSION THREAD

**Discussion by RockNHawk** [OrchardCMS/OrchardCore#7288] [Show and tell]
If you `ConfigureServices` Orchard Core with your custom config in main Application's Startup.cs , may show `Unable to resolve service for type 'YesSql.ISession' while attempting to activate 'OrchardCore.Workflows.Http.Services.HttpRequestRouteActivator' ` error in setup

add an `AddDataAccess()` call in  `ConfigureServices`  will resolve this issue.

	// copy & modified from AddOrchardCms
	var builder = services.AddOrchardCore()
			.AddCommands()
			.AddMvc()
			.AddEmailAddressValidator()
			.AddHtmlSanitizer()
			.AddSetupFeatures("OrchardCore.Setup")
			.AddDataAccess()
			.AddDataStorage()
			.AddBackgroundService()
			//.AddTheming()
			//.AddLiquidViews()
			.AddCaching()
		;

**Comment by hishamco** [comment]
Be carful the order is very important, furthermore the `ISession` is registered via `AddDataAccess()`. I'm not sure why It's needed in `HttpRequestRouteActivator` but let me check

**Reply by hishamco** [reply]
FYI the issue is already addressed [here](https://github.com/OrchardCMS/OrchardCore/issues/5352)

**Reply by hishamco** [reply]
BTW `HttpRequestRouteActivator` seems removed or renamed
